--- tripwire-1.2/src/config.parse.c
+++ tripwire-1.2/src/config.parse.c	1998/09/14 09:19:09
@@ -352,8 +352,14 @@
     /* clean up */
     if (!specified_configmode)
 	(void) fclose(fpin);
-
-    rewind(fpout);
+    else {
+       /* Bug fix:  fpin and fpout are the same. fpout was being */
+       /* rewound after fpin was being closed.  The behavior of  */
+       /* this is undefined, and causes core dumps occasionally. */
+       /* Added the else and changed fpout to fpin for clarity.  */
+       /* Bug Fix by Dan D Niles (dniles@iname.com) -DDN- */
+       rewind(fpin);
+    }
     return;
 }
