--- tripwire-1.2/configs/conf-linux.h.sopwith	Thu Jul 21 20:14:46 1994
+++ tripwire-1.2/configs/conf-linux.h	Fri Feb  7 02:51:42 1997
@@ -27,30 +27,30 @@
  * does your system have a <malloc.h> like System V? 
  */
 
-#undef MALLOCH 	
+#define MALLOCH 	1
 
 /* 
  * does your system have a <stdlib.h> like POSIX says you should? 
  */
 
-#undef STDLIBH
+#define STDLIBH		1
 
 /*
  * does your system use readdir(3) that returns (struct dirent *)?
  */
 
-#undef DIRENT
+#define DIRENT		1
 
 /*
  * is #include <string.h> ok?  (as opposed to <strings.h>)
  */
 
-#define STRINGH
+#define STRINGH		1
  
 /* 
  * does your system have gethostname(2) (instead of uname(2))?
  */
 
-#define GETHOSTNAME
+#define GETHOSTNAME	1
 
 #define TW_LINUX
--- tripwire-1.2/include/config.h.sopwith	Fri Jul 15 07:02:52 1994
+++ tripwire-1.2/include/config.h	Fri Feb  7 02:51:42 1997
@@ -17,7 +17,7 @@
  ***	file that corresponds with your operating system.
  ***/
 
-#include "../configs/conf-svr4.h"
+#include "../configs/conf-linux.h"
 
 #ifdef TW_TYPE32
 typedef TW_TYPE32 int32;
@@ -103,8 +103,8 @@
 #endif
 */
 
-#define CONFIG_PATH     "/tmp/genek"
-#define DATABASE_PATH   "/tmp/genek"
+#define CONFIG_PATH     "/etc"
+#define DATABASE_PATH   "/var/spool/tripwire"
 
 /******* name of Tripwire files **************************************
  *
@@ -162,7 +162,4 @@
  *
  *********************************************************************/
 
-#define TEMPFILE_TEMPLATE "/tmp/twzXXXXXX"
-
-
-
+#define TEMPFILE_TEMPLATE "/var/tmp/.twzXXXXXX"
--- tripwire-1.2/lib/tw.config.sopwith	Mon Nov 22 01:38:06 1993
+++ tripwire-1.2/lib/tw.config	Fri Feb  7 02:51:42 1997
@@ -28,6 +28,26 @@
 #	u :  user id of owner			1: signature 1
 #	g :  group id of owner			2: signature 2
 #	s :  size of file
+# Modes we want:
+
+# Log file
+@@define LOGFILEM E+pugn
+
+# Config file
+@@define CONFM E+pinugc
+
+# Binary
+@@define BINM E+pnugsci12
+
+# Directory
+@@define DIRM E+pnug
+
+# Data file (same as BIN_M currently)
+@@define DATAM E+pnugsci12
+
+# Device files
+@@define DEVM E+pnugsci
+
 #
 #
 # Ex:   The following entry will scan all the files in /etc, and report
@@ -90,41 +110,35 @@
 #		@@endif
 #
 
-/tmp/genek/tripwire-1.0				R+0123456789-ugpmci
-/tmp/genek/tripwire-1.0/tests 			E
-/tmp/genek/tripwire-1.0/tw.config	 	E
-/tmp/genek/tripwire-1.0/databases 		E
-
-/tmp/genek/tripwire-1.0/src/config.parse.o
-/tmp/genek/tripwire-1.0/src/main.o
-/tmp/genek/tripwire-1.0/src/list.o
-/tmp/genek/tripwire-1.0/src/ignorevec.o
-/tmp/genek/tripwire-1.0/src/dbase.build.o
-/tmp/genek/tripwire-1.0/src/utils.o
-/tmp/genek/tripwire-1.0/src/preen.o
-/tmp/genek/tripwire-1.0/src/preen.interp.o
-/tmp/genek/tripwire-1.0/src/preen.report.o
-/tmp/genek/tripwire-1.0/src/nullsig.o
-/tmp/genek/tripwire-1.0/src/config.prim.o
-/tmp/genek/tripwire-1.0/src/dbase.update.o
-/tmp/genek/tripwire-1.0/src/sigfetch.o
-/tmp/genek/tripwire-1.0/src/config.pre.o
-/tmp/genek/tripwire-1.0/sigs/crc/crc.o
-/tmp/genek/tripwire-1.0/sigs/crc32/crc32.o
-/tmp/genek/tripwire-1.0/sigs/md2/md2wrapper.o
-/tmp/genek/tripwire-1.0/sigs/md2/md2.o
-/tmp/genek/tripwire-1.0/sigs/md4/md4.o
-/tmp/genek/tripwire-1.0/sigs/md4/md4wrapper.o
-/tmp/genek/tripwire-1.0/sigs/md5/md5.o
-/tmp/genek/tripwire-1.0/sigs/md5/md5wrapper.o
-/tmp/genek/tripwire-1.0/sigs/snefru/snefru.o
-
-!/tmp/genek/tripwire-1.0/aux/types
-!/tmp/genek/tripwire-1.0/aux/byteorder
-
-!/tmp/genek/tripwire-1.0/src/config.lex.c
-!/tmp/genek/tripwire-1.0/src/config.pre.c
-!/tmp/genek/tripwire-1.0/Makefile
-!/tmp/genek/tripwire-1.0/include/config.h
-!/tmp/genek/tripwire-1.0/include/byteorder.h
-!/tmp/genek/tripwire-1.0/include/inode.h
+# exclude all of /proc
+=/proc E
+
+
+#=/dev @@DIRM
+/dev @@DEVM
+
+#=/etc @@DIRM
+/etc @@CONFM
+
+# Binary directories
+#=/usr/sbin @@DIRM
+/usr/sbin @@BINM
+
+#=/usr/bin @@DIRM
+/usr/bin @@BINM
+
+#=/sbin @@DIRM
+/sbin @@BINM
+
+#=/bin @@DIRM
+/bin @@BINM
+
+#=/lib @@DIRM
+/lib @@BINM
+
+#=/usr/lib @@DIRM
+/usr/lib @@BINM
+
+=/usr/src E
+
+=/tmp @@DIRM
--- tripwire-1.2/src/Makefile.sopwith	Mon Jul 25 11:40:54 1994
+++ tripwire-1.2/src/Makefile	Fri Feb  7 02:51:42 1997
@@ -104,7 +104,7 @@
 	$(CC) $(CFLAGS) -c $<
 
 install:	tripwire
-	$(INSTALL) tripwire $(DESTDIR)
+	$(INSTALL) -s -m700 siggen tripwire $(DESTDIR)
 
 clean:
 	-rm -f $(OFILES) config.lex.c config.pre.c y.tab.c lex.yy.c help.c \
--- tripwire-1.2/src/config.pre.y.sopwith	Fri Feb  7 02:52:27 1997
+++ tripwire-1.2/src/config.pre.y	Fri Feb  7 02:52:56 1997
@@ -63,7 +63,7 @@
 extern FILE *yyin;
 extern FILE *yyout;
 
-#ifdef TW_LINUX
+#ifdef TW_LINUX_UNDEF
 #include <malloc.h>
 
 	void *yy_flex_realloc(void *x,int y) { return realloc(x,y); }
--- tripwire-1.2/Makefile.sopwith	Mon Jul 25 11:59:41 1994
+++ tripwire-1.2/Makefile	Fri Feb  7 02:51:42 1997
@@ -12,14 +12,17 @@
 ###
 
 # destination directory for final executables
-DESTDIR = /secureplace/bin
+# This is a joke anyways to have a tripwire rpm, but people want it,
+# so stick things in standard places
+ROOTDIR=/
+DESTDIR = $(TOPDIR)/usr/sbin
 
 # destination for man pages
-MANDIR  = /usr/man
+MANDIR  = $(TOPDIR)/usr/man
 
 # system utilities
-LEX	= lex
-#LEX	= flex			# For the GNU crowd
+#LEX	= lex
+LEX	= flex			# For the GNU crowd
 
 YACC	= yacc
 #YACC	= bison -y		# For the GNU crowd (make it look like yacc)
@@ -35,8 +38,8 @@
 CC     = cc			# common
 #CC     = gcc			# also common
 #CC	= /usr/ccs/bin/cc	# Pyramid DC/OSx (SVR4)
-
-CFLAGS = -O			# common 
+CFLAGS=$(RPM_OPT_FLAGS)
+#CFLAGS = -O			# common 
 #CFLAGS = -g 			# common 
 #CFLAGS = -g			# debugging
 #CFLAGS = -O -cckr		# SGI
@@ -60,7 +63,7 @@
 #CPP	= /lib/cpp		# on older systems
 
 # make sure libraries are not linked dynamically (as a security measure)
-LDFLAGS=			# common
+LDFLAGS=-static			# common
 #LDFLAGS= -non_shared		# OSF/1
 #LDFLAGS= -Bstatic		# SunOS 4 (cannot statically link tripwire
                           	#          on Solaris 2.3)
